cfn-resources/stream-connection/cmd/resource/resource.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+)
+
+const (
+	ClusterConnectionType = "Cluster"
+	KafkaConnectionType   = "Kafka"
+)
+
+var CreateRequiredFields = []string{constants.ProjectID, constants.InstanceName, constants.ConnectionName, constants.Type}
+var ReadRequiredFields = []string{constants.ProjectID, constants.InstanceName, constants.ConnectionName}
+var UpdateRequiredFields = []string{constants.ProjectID, constants.InstanceName, constants.ConnectionName, constants.Type}
+var DeleteRequiredFields = []string{constants.ProjectID, constants.InstanceName, constants.ConnectionName}
+var ListRequiredFields = []string{constants.ProjectID, constants.InstanceName}
+
+func setup(cfnFunc constants.CfnFunctions, req handler.Request, currentModel *Model) (*admin.APIClient, *handler.ProgressEvent) {
```

Comment:
I have renamed it to `initEnvWithLatestClient` for now but open to suggestions.
I am not sure if this can be made reusable since "Model" object differs for each resource package, but I am open to suggestions. Either ways we can add this to the boilerplate code in autogen

Reply:
ahh true. good point regarding the autogen